src/OcelotGateway/Services/GatewayConfigurationService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Middleware.Common.MessageContracts;
+using Yarp.ReverseProxy.Configuration;
+using Yarp.ReverseProxy.Transforms;
+
+namespace Middleware.OcelotGateway.Services;
+
+public class GatewayConfigurationService
+{
+    private readonly InMemoryConfigProvider _inMemoryConfigProvider;
+
+    public GatewayConfigurationService(IProxyConfigProvider inMemoryConfigProvider)
+    {
+        if (inMemoryConfigProvider is InMemoryConfigProvider imcp)
+            _inMemoryConfigProvider = imcp;
+    }
+
+    public void CreateDynamicRoute(GatewayAddNetAppEntryMessage msg)
+    {
+        var config =  _inMemoryConfigProvider.GetConfig();
+
+        var clusterList = config.Clusters.ToList();
+        var routeList = config.Routes.ToList();
+
+        var clusterCfg = new ClusterConfig
+        {
+            ClusterId = msg.NetAppName + "-Cluster",
+            Destinations = new Dictionary<string, DestinationConfig>
+            {
+                { "destination1", new DestinationConfig { Address = $"http://{msg.NetAppName}" } }
+            }
+        };
+        var routeCfg = new RouteConfig
+        {
+            RouteId = msg.NetAppName + "-Route",
+            Match = new()
+            {
+                Path = msg.NetAppName
+            },
+            ClusterId = clusterCfg.ClusterId //
+        };
+        // transforms allow us to change the path that is requested like below to replace direct forwarding
+        routeCfg = routeCfg.WithTransformPathSet("/");
```

Comment:
```suggestion
        routeCfg = routeCfg.WithTransformPathRemovePrefix($"/{msg.NetAppName}");
```
I have read a bit more on the Transforms, and it turned out that the prefix removal will be a better option for us. Per documentation we're changing `/prefix/request/path` to `/request/path`, which is the desired action. We want only to trim the beginning and not replace a whole path. This will keep the remainder of a request and pass it to the NetApp.